File: spring-webflux/webflux/r2dbc/src/main/java/com/example/webflux/r2dbc/controller/dto/UserResponse.java
package com.example.webflux.r2dbc.controller.dto;

import com.example.webflux.r2dbc.common.User;
import lombok.Data;

import java.util.Optional;

@Data
public class UserResponse {
    private final String id;
    private final String name;
    private final int age;
    private final Long followCount;
    private final Optional<ProfileImageResponse> image;

    public static UserResponse of(User user) {
        return new UserResponse(
                user.getId(),
                user.getName(),
                user.getAge(),
                user.getFollowCount(),
                user.getProfileImage().map(image ->
                        new ProfileImageResponse(
                                image.getId(),
                                image.getName(),
                                image.getUrl()))
        );
    }
}

File: spring-webflux/webflux/r2dbc-coroutine/src/main/kotlin/com/example/webflux/r2dbc/controller/dto/UserResponse.kt
package com.example.webflux.r2dbc.controller.dto

import com.example.webflux.r2dbc.common.Image
import com.example.webflux.r2dbc.common.User
import java.util.Optional

data class UserResponse(
    val id: String,
    val name: String,
    val age: Int,
    val followCount: Long,
    val image: Optional<ProfileImageResponse>
) {
    companion object {
        fun of(user: User): UserResponse {
            return UserResponse(
                user.id,
                user.name,
                user.age,
                user.followCount,
                user.profileImage.map { image: Image ->
                    ProfileImageResponse(
                        image.id,
                        image.name,
                        image.url
                    )
                }
            )
        }
    }

}